docs/blog/2018-11-03-building-an-accessible-lightbox/index.md

```diff
@@ -0,0 +1,348 @@
+
+```js
+{
+  resolve: `gatsby-source-filesystem`,
+  options: {
+    name: `cars`,
+    path: `${__dirname}/src/cars`,
+  },
+},
+```
+
+Add that under the existing filesystem config and then restart your Gatsby server (I believe it's necessary for it to build the query before calling it).
+
+### Creating a `Cars` component
+
+Now that we have configured the filesystem, we can go ahead and create a component that will request all the images with a GraphQL query.
+
+Go ahead and create a new file inside `src/components` called `cars.js`.
+
+```js
+import React from "react"
+import { StaticQuery, graphql } from "gatsby"
+import Lightbox from "./lightbox"
+
+const Cars = () => (
+  <StaticQuery
+    query={graphql`
+      query {
+        carImages: allFile(filter: { sourceInstanceName: { eq: "cars" } }) {
+          edges {
+            node {
+              childImageSharp {
+                fluid(maxWidth: 2000) {
+                  ...GatsbyImageSharpFluid
+                }
+              }
+            }
+          }
+        }
+      }
+    `}
+    render={data => <Lightbox carImages={data.carImages.edges} />}
+  />
+)
+export default Cars
+```
+
+Notice how the `StaticQuery` component is using a render prop where we're returning our `Lightbox` component. For now it's a simple component inside `components/lightbox.js` that looks like this:
+
+```js
+import React, { Component } from "react"
+import PropTypes from "prop-types"
+import Img from "gatsby-image"
+
+export default class Lightbox extends Component {
+  static propTypes = {
+    carImages: PropTypes.array.isRequired,
+  }
+
+  render() {
+    const { carImages } = this.props
+    return (
+      <div>
+        {carImages.map(image => (
+          <Img
+            key={image.node.childImageSharp.fluid.src}
+            fluid={image.node.childImageSharp.fluid}
+          />
+        ))}
+      </div>
+    )
+  }
+}
+```
+
+By now, we should have all the images displaying on the home page and doing a fade-up load.
+
+### Styled Components
+
+For styling, we're going to be using [styled-components](https://www.styled-components.com/). To get it configured with Gatsby, run the following inside your terminal in our application:
+
+```terminal
+$ yarn add gatsby-plugin-styled-components styled-components babel-plugin-styled-components
+```
+
+And add the following to `gatsby-config.js`:
+
+```js
+module.exports = {
+  /* existing config */
+  plugins: [
+    `gatsby-plugin-styled-components`,
+    /* existing plugins */
+  ],
+}
+```
+
+Now, for the sake of consistency, lets restart the application. Now, we'll create a simple `LightboxContainer` styled component to wrap our images in. Your `lightbox.js` component should look like this now:
```

Comment:
Just for the sake of consistency?

Reply:
@shannonbux Hey Shannon, thanks so much for your feedback and suggestions. I've updated the blog accordingly. Hopefully I didn't miss anything 😅Sorry in advance if I did!